server/src/main/java/org/apache/druid/client/JsonParserIterator.java

```diff
@@ -52,12 +56,14 @@
   private final String host;
   private final ObjectMapper objectMapper;
   private final BytesAccumulatingResponseHandler responseHandler;
+  private final boolean hasTimeout;
+  private final long timeoutAt;
 
   public JsonParserIterator(
       JavaType typeRef,
       Future<InputStream> future,
       String url,
-      Query<T> query,
+      @Nullable Query<T> query,
```

Comment:
Would you please add `@Nullable` to the member variable `query` as well? Then Intellij will perform a proper inspection based on the annotation.